code/utils/subxt-client/default.nix

```diff
@@ -0,0 +1,96 @@
+{ pkgs, subxt }:
+let
+  registriesConf = pkgs.writeText "registries.conf" ''
+    [registries.search]
+    registries = ['docker.io']
+    [registries.block]
+    registries = []
+  '';
+
+  containersConf = pkgs.writeText "containers.conf" ''
+    [containers]
+    default_capabilities = [
+        "CHOWN",
+        "DAC_OVERRIDE",
+        "FOWNER",
+        "FSETID",
+        "KILL",
+        "NET_BIND_SERVICE",
+        "SETFCAP",
+        "SETGID",
+        "SETPCAP",
+        "SETUID",
+        "SYS_CHROOT"
+    ]
+    default_sysctls = [
+        "net.ipv4.ping_group_range=0 0",
+    ]
+    [secrets]
+    [secrets.opts]
+    [network]
+    [engine]
+    [engine.runtimes]
+    [engine.volume_plugins]
+    [machine]
+  '';
+
+  tools = pkgs.writeText "tools.sh" ''
+    wait_for_ws_server_to_be_ready() {
+        HOST=$1
+        PORT=$2
+        set +e
+        while true; do
+            echo "Trying to connect to $HOST:$PORT..."
+            echo '{"id":1, "jsonrpc":"2.0", "method": "system_version"}' | websocat --oneshot ws://$HOST:$PORT > /dev/null
+            exit_code=$?
+            if [ $exit_code -eq 0 ]; then
+                break
+            fi
+            sleep 1
+        done
+        set -e
+    }
+  '';
+
+in pkgs.stdenv.mkDerivation {
+  name = "subxt-client";
+  dontUnpack = true;
+  buildInputs = [
+    pkgs.podman
+    pkgs.runc # Container runtime
+    pkgs.conmon # Container runtime monitor
+    pkgs.skopeo # Interact with container registry
+    pkgs.slirp4netns # User-mode networking for unprivileged namespaces
+    pkgs.fuse-overlayfs # CoW for images, much faster than default vfs
+    pkgs.websocat
+    pkgs.rustfmt
+    subxt
+  ];
+  installPhase = ''
+    mkdir $out
+    export HOME=$(pwd)
+    mkdir --parent $out/etc/containers $HOME/.config/containers
+
+    ln -s ${pkgs.skopeo.src}/default-policy.json $out/etc/containers/policy.json
+    ln -s ${registriesConf} $out/etc/containers/registries.conf
+    ln -s ${containersConf} $HOME/.config/containers/containers.conf
+    source ${tools}
+
+    podman version
+
+    container_id=$(podman run --rm -d -u$(id -u):$(id -g) -p9944:9944 -p9988:9988 docker.io/composablefi/composable-sandbox:latest)
```

Comment:
so you are open ports anyway? and access podman from nix build to ports opened?

Reply:
why then not run just pure binaries?

Reply:
idea is that you run nodes AND generator insited podman so you no need to open ports. if you have to open ports with podman, than can just run nodes without isolation then?

Reply:
so you
1. cp into podman node and subxtclient
2. run node
3. run client 
4. cp back output